### PR TITLE
Move all quantity related auxiliary functions out of Maps::Tiles class

### PR DESCRIPTION
**Pull request by ihhub**
To standartize the code for the future changes and to be able to remove cryptic quantity logic.

relates to #6845

### CODE REVIEW

**Review by oleg-derevenetz — COMMENTED** [review]
Hi @ihhub I just left one question, could you please take a look when you have time?

File: src/fheroes2/maps/maps_tiles_helper.cpp [hunk 1]
```diff
@@ -0,0 +1,1315 @@
...
+        return Funds( rc );
+    }
+
+    Troop getTroopFromTile( const Tiles & tile )
+    {
+        return MP2::isCaptureObject( tile.GetObject( false ) ) ? world.GetCapturedObject( tile.GetIndex() ).GetTroop()
+                                                               : Troop( getMonsterFromTile( tile ), getMonsterCountFromTile( tile ) );
+    }
+
+    int getColorTypeFromTile( const Tiles & tile )
```
[the remark this answers is not in the document]
Comment (ihhub) [reply src/fheroes2/maps/maps_tiles_helper.cpp:424]: Mmm... no idea what my intention was but I will change it.

File: src/fheroes2/maps/maps_tiles_helper.cpp [hunk 2]
```diff
@@ -0,0 +1,1315 @@
...
+                    tile.QuantitySetVariant( 15 );
+                    setSpellOnTile( tile, spell );
+                }
+                else {
+                    // 0: 70% none
+                    // 1,2,3 - 2000g, 2500g+3res, 3000g+5res,
+                    // 4,5 - need to have skill wisdom or leadership,
+                    // 6 - 50 rogues, 7 - 1 gin, 8,9,10,11,12,13 - 1 monster level4,
+                    // 15 - spell
+                    int cond = Rand::Get( 1, 10 ) < 4 ? Rand::Get( 1, 13 ) : 0;
```
Comment (Districh-ru) [inline src/fheroes2/maps/maps_tiles_helper.cpp:717]: This variable can be **const**, as in line 970.

File: src/fheroes2/maps/maps_tiles_helper.cpp [hunk 3]
```diff
@@ -0,0 +1,1315 @@
...
+            // 40% - 10ghost(1000g)
+            percents.Push( 1, 40 );
+            // 30% - 15 ghost(2000g)
+            percents.Push( 2, 30 );
+            // 20% - 25ghost(5000g)
+            percents.Push( 3, 20 );
+            // 10% - 50ghost(2000g+art)
+            percents.Push( 4, 10 );
+
+            int cond = percents.Get();
```
Comment (Districh-ru) [inline src/fheroes2/maps/maps_tiles_helper.cpp:948]: This one also can be **const**.

File: src/fheroes2/maps/maps_tiles_helper.cpp [hunk 4]
```diff
@@ -0,0 +1,1315 @@
...
+        static_assert( std::is_same_v<TileImageIndexType, uint8_t>, "Type of GetObjectSpriteIndex() has been changed, check the logic below" );
+
+        assert( mons.GetID() > std::numeric_limits<TileImageIndexType>::min() && mons.GetID() <= std::numeric_limits<TileImageIndexType>::max() );
+
+        tile.setObjectSpriteIndex( static_cast<TileImageIndexType>( mons.GetID() - 1 ) ); // ICN::MONS32 starts from PEASANT
+    }
+
+    void updateMonsterInfoOnTile( Tiles & tile )
+    {
+        Monster mons = Monster( tile.GetObjectSpriteIndex() + 1 ); // ICN::MONS32 start from PEASANT
```
Comment (Districh-ru) [inline src/fheroes2/maps/maps_tiles_helper.cpp:1249]: This variable is not changed and can be **const**.

**Review by Districh-ru — APPROVED** [review]
Hi, @ihhub.
I checked this PR by loading some maps and saves and did not find any issues.
The code becomes more easier to read!
I just have a couple of proposals which you can address if you think it's necessary.

**Review by oleg-derevenetz — APPROVED** [review]

**Comment by ihhub** [comment]
Hi @Districh-ru , I addressed all given comments from you.

**Review by Districh-ru — APPROVED** [review]

**Pull request merged**